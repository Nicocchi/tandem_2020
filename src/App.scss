.App {
  height: 100%;
  text-align: center;
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  height: calc(100% - 6vh);
  padding: 20px;

  img {
    width: 100%;
    max-width: 600px;
  }
}

.play-btn {
  margin-top: 40px;
  font-size: 3em;
  // color: #cc00f8;
  color: #33c7f5;
  cursor: pointer;

  &:hover {
    // color: #551a8b;
    color: #208dea;
  }
}

// Header
.game-header {
  height: 50px;
  display: flex;
  justify-content: flex-end;
  padding-right: 10px;

  .answer-btn {
    height: 30px;
    width: 80px;
    font-size: 1em;
  }
}

// Game
.game-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: calc(100% - 6vh);
}

// QuestionBox
.question {
  font-size: 1.7em;
  color: #fff;
}

.answers-box {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.answer-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.answer-btn {
  border: none;
  width: 300px;
  height: 50px;
  border-radius: 20px;
  margin-top: 10px;
  background-color: #2678e6;
  color: #fff;
  font-family: 'Luckiest Guy', cursive;
  font-size: 1.4em;
  padding-top: 6px;
  cursor: pointer;

  &:hover {
    background-color: #4031d4;
  }
}

.next-btn {
  background-color: #12d239;
  margin-top: 40px;
  cursor: pointer;

  &:hover {
    background-color: #0faa2e;
  }
}

.answer-text {
  display: flex;
  justify-content: space-evenly;

  p {
    font-size: 1.4em;
    padding: 5px;
  }
}

.incorrect-text {
  color: red;
}

.correct-text {
  color: #12d239;
}

// Score
.score-box {
  p, h1 {
    color: #fff;
  }

  h1 {
    font-size: 4em;
  }

  p {
    font-size: 1.4em;
    padding: 0 5px 0 5px;
  }

  span {
    display: flex;
    justify-content: center;
  }
}

.score-num {
  color: #0faa2e !important;
}

// Animations
.animated { 
  animation-duration: 2s; 
  animation-fill-mode: both; 
} 

.animated-infinite { 
  animation-duration: 2s; 
  animation-fill-mode: both; 
  animation-iteration-count: infinite;
  -webkit-animation-iteration-count: infinite;
} 

@keyframes bounce-in-right { 
  0% { 
      opacity: 0; 
      transform: translateX(2000px); 
  } 
  60% { 
      opacity: 1; 
      transform: translateX(-30px); 
  } 
  80% { 
      transform: translateX(10px); 
  } 
  100% { 
      transform: translateX(0); 
  } 
} 

.bounce-in-right { 
  animation-name: bounce-in-right; 
}

@keyframes fade-in { 
  0% {opacity: 0;} 
  100% {opacity: 1;} 
} 
.fade-in { 
  animation-name: fade-in; 
}

@keyframes flip-in-y { 
  0% { 
      transform: perspective(400px) rotateY(90deg); 
      opacity: 0; 
  } 
  40% { 
      transform: perspective(400px) rotateY(-10deg); 
  } 
  70% { 
      transform: perspective(400px) rotateY(10deg); 
  } 
  100% { 
      transform: perspective(400px) rotateY(0deg); 
      opacity: 1; 
  } 
} 
.flip-in-y { 
  backface-visibility: visible !important; 
  animation-name: flip-in-y; 
}

@keyframes swing { 
  20% { transform: rotate(15deg); } 
  40% { transform: rotate(-10deg); } 
  60% { transform: rotate(5deg); } 
  // 80% { transform: rotate(-5deg); } 
  80% { transform: rotate(0deg); } 
  100% { transform: rotate(0deg); } 
} 
.swing { 
  transform-origin: top center; 
  animation-name: swing; 
}

@keyframes bounce-in-down { 
  0% { 
      opacity: 0; 
      transform: translateY(-2000px); 
  } 
  60% { 
      opacity: 1; 
      transform: translateY(30px); 
  } 
  80% { 
      transform: translateY(-10px); 
  } 
  100% { 
      transform: translateY(0); 
  } 
} 

.bounce-in-down { 
  animation-name: bounce-in-down; 
}

// Misc
.title1 {
  color: #d000ff;
  font-size: 6em;
}

.title2 {
  font-size: 4em;
}

.text-shadow-pop-left {
  margin: 0;
	-webkit-animation: text-shadow-pop-left 0.6s both;
	        animation: text-shadow-pop-left 0.6s both;
}

@-webkit-keyframes text-shadow-pop-left {
  0% {
    text-shadow: 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097;
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  100% {
    text-shadow: -1px 0 #7b0097, -2px 0 #7b0097, -3px 0 #7b0097, -4px 0 #7b0097, -5px 0 #7b0097, -6px 0 #7b0097, -7px 0 #7b0097, -8px 0 #7b0097;
    -webkit-transform: translateX(8px);
            transform: translateX(8px);
  }
}
@keyframes text-shadow-pop-left {
  0% {
    text-shadow: 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097, 0 0 #7b0097;
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  100% {
    text-shadow: -1px 0 #7b0097, -2px 0 #7b0097, -3px 0 #7b0097, -4px 0 #7b0097, -5px 0 #7b0097, -6px 0 #7b0097, -7px 0 #7b0097, -8px 0 #7b0097;
    -webkit-transform: translateX(8px);
            transform: translateX(8px);
  }
}
