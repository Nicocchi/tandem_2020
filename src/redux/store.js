import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import combineReducer from "./reducers";

let store = null;

// Redux dev tools
const composeEnhancers =
    typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
              // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
          })
        : compose;

const devEnhancer = composeEnhancers(applyMiddleware(thunk, logger));
const enhancer = composeEnhancers(applyMiddleware());

// Create redux store
if (process.env.REACT_APP_NODE_ENV !== "production") {
    store = createStore(combineReducer, devEnhancer);
} else {
    store = createStore(combineReducer, enhancer);
}

export default store;