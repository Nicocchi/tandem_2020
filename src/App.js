import React, { useEffect } from "react";
import './App.css';
import { getQuestions } from './redux/actions';
import { connect } from "react-redux";
import { Switch, withRouter } from "react-router-dom";
import routes from "./Routes";

function App(props) {
  const { getQuestions } = props;
  useEffect(() => {
    getQuestions();
    return () => {
    }
  }, [getQuestions]);

  return (
    <div className="App">
      <Switch>
        {routes}
      </Switch>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    questions: state.gameReducer.questions,
  }
}

export default connect(mapStateToProps, {getQuestions})(withRouter(App));
