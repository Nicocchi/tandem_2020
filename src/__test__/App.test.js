import React from "react";
import ReactDOM from "react-dom";
import App from "./../App";
import store from "./../redux/store";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

describe("App", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });

    afterEach(() => {
        cleanup();
    });

    it("renders without crashing without props", () => {
        const div = document.createElement("div");

        ReactDOM.render(
            <Provider store={store}>
                <Router history={history}>
                    <App history={history} />
                </Router>
            </Provider>,
            div
        );
    });

    it("matches snapshop", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Router history={history}>
                        <App history={history} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
