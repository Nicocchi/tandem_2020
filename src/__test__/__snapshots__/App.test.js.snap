// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshop 1`] = `
<div
  className="App"
>
  <div
    className="game-header"
  />
  <header
    className="App-header"
  >
    <p
      className="text-shadow-pop-left title1"
    >
      Tandem
    </p>
    <p
      className="text-shadow-pop-left title2"
    >
      Trivia
    </p>
    <a
      aria-current={null}
      className="play-btn animated-infinite swing"
      href="/play"
      onClick={[Function]}
    >
      Play
    </a>
  </header>
</div>
`;
